File: User Services/freelancer/src/main/java/com/worksphere/freelancer/entity/Requirement.java
package com.worksphere.freelancer.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "requirement")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Requirement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "requirement_id")
    private Integer requirementId;

    @ManyToOne
    @JoinColumn(name = "client_id")
    private User client;

    @ManyToOne
    @JoinColumn(name = "service_id")
    private Services service;

    @Column(name = "description")
    private String description;

    @Column(name = "created_date")
    private LocalDate createdDate;

    @Column(name = "deadline")
    private LocalDate deadline;

    @Column(name = "experience")
    private Integer experience;

    @Column(name = "budget")
    private BigDecimal budget;

    @Column(name = "status")
    private String status;
}

File: RestAPIs/Admin_Services/Models/Requirement.cs
using System;
using System.Collections.Generic;

namespace Admin_Services.Models;


public partial class Requirement
{
    public int RequirementId { get; set; }

    public int ClientId { get; set; }

    public int ServiceId { get; set; }

    public string? Description { get; set; }

    public DateOnly? CreatedDate { get; set; }

    public string? Status { get; set; }

    public virtual User Client { get; set; } = null!;

    public virtual ICollection<RequirementRequest> RequirementRequests { get; set; } = new List<RequirementRequest>();

    public virtual Service Service { get; set; } = null!;
}
